swift/ios-shoppe-demo/ServiceManagers/FullStoryManager.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  FullStoryManager.swift
+//  ios-shoppe-demo
+//
+//  Created 5/7/20.
+//  Copyright © 2020 FullStory All rights reserved.
+//
+
+import Foundation
+import FullStory
+
+/**
+ FullStory Manager is a singleton used to manage api calls to FullStory.
+
+    - create events
+    - identify a user masking and unmasking elements
+ */
+class FullStoryManager {
```

Comment:
Overall this is a great set of functionality.  I'm kinda surprised it's not in core code, but there's the opportunity.  What about making some of the more generic convenience methods an `Extension` to `FS`?  Is that possible?  So you'd take out ios-shoppe specific functionality and create a common extension that could be used by anyone.

It also seems like a lot of the Manager's functions pass though parameters in a more convenient way.  As such, you could have static functions rather than a singleton.

Reply:
I think to 2 key conveniences here are:

`enum LogLevel {
    case assert, error, warning, info, debug
}

enum FSStatus: String {
    case exclude = "fs-exclude"
    case mask = "fs-mask"
    case unmask = "fs-unmask"
}`

...as well as the methods `createFSEvent` and `log`

Maybe I will remove them from this singleton since there is no good reason yet for it to be this way. The original reason is it would have a `currentCustomEvent` to hold the status of where the user is in the app, it would have a boolean value for `userConsentGiven`, etc...

For now I am abstracting the 4 items out of a class as they are something that can be copy/pasted into a given project.

- LogLevel
- FSStatus

- fsCreateEvent
- fsLog
- fsModify